### DISCUSSION TITLE
KeyFactor/EJBCA-CE (8.0) not giving public web pages

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#376] [General]
Hi guys, I deployed EJBCA 8.0 image (github) via docker with external mysql DB with TLS setup enable to true. Now I am trying to get public web pages _(which we usually get pre 8.0 versions)_ without client authentication but not getting them. So can anyone help how can I get them on port 443 without client authentication.

**NOTE:** I want to keep client authentication enabled with same port 443 on adminweb,RA & ejbcaws directories.

Public web without client authentication is required only for user enrollment purposes.

I ran following docker command to build my container.

docker run -it -p 80:8080 -p 443:8443 --name prod -h pki-apps.abc.com -e TLS_SETUP_ENABLED="true" -e ADMINWEB_ACCESS="true" -e DATABASE_JDBC_URL="jdbc:mariadb://10.20.1.5:6001/proddb?characterEncoding=UTF-8" - e DATABASE_USER="prod" -e DATABASE_PASSWORD="test" keyfactor/ejbca-ce

**Comment by svenska-primekey** [comment]
Public web is deprecated, try accessing adminweb, raweb.

**Comment by merryo11** [comment]
Dear Sven,  I am using admin web and raweb but the problem is they both
require client authentication certificate but we cannot give client
authentication certificate to everyone general user for just enrollment
purposes.

**Reply by svenska-primekey** [reply]
The container logs will have the password for creating the credential. You should still be able to get to the ra web to create the credential using the use username to create the p12.

Otherwise you could drop the DB tables and relaunch the container with the env `TLS_SETUP_ENABLED=simple` which would let you in to create a credential and then you can lock down EJBCA access.

**Comment by merryo11** [comment]
Well Sven, by creating new container and dropping the tables is not a
solution for me because I already created few CAs, cert and end entity
profiles so I shall lose all that data and rework it.

**Reply by svenska-primekey** [reply]
Then you can use the CLI to exec into the container and create the p12 file. Then copy it out of the container.

**Reply by merryo11** [reply]
Well I think you are thinking that I am asking for superadmin certificate so this is not the case. See my very first post where I already mentioned that my container got setup perfectly with external DB and i downloaded the superadmin also and my adminweb and RA portals are working fine. 

I am just asking about what link shall I share with my general users for just certificate enrollment using username in p12 format. I cannot give admin certificate to everyone as they are supposed to be with nominated RA (Registration Authority) personals only.

If there is no possibility then my RA can create the entities and generate end user certificate and send them via email or via some other mechanism for their use.

My users are just using the normal end entity certificates for different roles like accessing HR/Finance portals through digital certificates and local email signing and document signing purposes.

**Reply by primetomas** [reply]
The RA web can be used by unatheenticated users. It's somewhere in the documentation...
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ra-concept-guide

When creating roles there is something called a PublicAccessToken that you can use to allow unathenticated access to the Admin UI.
I think when starting a (new) container initially, with TLS_SETUP_ENABLED=true, the link provided in the log to give you the initial superadmin certificate will use this.

So the same URL that you used to get your initial superadmin ceritficate can be used for other certificates.

**Reply by kumareshsomi** [reply]
Indeed as Tomas pointed out, the answer is in here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ra-operations-guide/configure-ejbca-for-public-access

**Reply by merryo11** [reply]
Dear I followed Kumar link and I got the pages over https and without any admin certificate. Now the problem that I am facing is that I am not able to complete the enrollment using username and getting error that the entity doesn't not exist (see screen shot below).  When I do that same thing from RA portal within adminweb the same entity gets enrolled. Don't know why it this is happening. Is there some right issues.

My access link to the portal is https://10.20.1.2/ejbca/ra/enrollusername.xhtml
![userenroll-error](https://github.com/Keyfactor/ejbca-ce/assets/144104956/2db1c6bc-6cc3-4929-9bf1-b5141c110421)

**Reply by primetomas** [reply]
This is not the same issue. Please start a new discussion.
